### PR TITLE
Allow to receive any native message

### PR DESCRIPTION
**Pull request by mauroservienti**
Even if there are no NServiceBus known headers

### CODE REVIEW

**Comment by mauroservienti** [comment]
Currently, it blows up here https://github.com/Particular/NServiceBus.AmazonSQS/blob/866a007cc75ea8d92d087d8d3d4bd542df7423e5/src/NServiceBus.Transport.SQS/InputQueuePump.cs#L336

When there are no known headers the transport assumes that on the wire there a json-serialized `TransportMessage`.

**Comment by mikeminutillo** [comment]
Added an attempt at making this test pas in 13eb3351f6c2635c67a92a243b3f64fd20d2d477

File: src/NServiceBus.Transport.SQS/InputQueuePump.cs [hunk 1]
```diff
@@ -282,8 +282,10 @@ internal async Task ProcessMessage(Message receivedMessage, CancellationToken me
             {
                 try
                 {
+                    var nativeHeadersContainNServiceBusMessageIdHeader = false;
                     if (receivedMessage.MessageAttributes.TryGetValue(Headers.MessageId, out var messageIdAttribute))
                     {
+                        nativeHeadersContainNServiceBusMessageIdHeader = true;
```
Comment (mauroservienti) [inline src/NServiceBus.Transport.SQS/InputQueuePump.cs:288]: TODO: validate if checking for the presence of the message ID in the _native_ headers is enough to determine the sender is an NServiceBus endpoint. Are all versions of the transport setting the header?

File: src/NServiceBus.Transport.SQS/InputQueuePump.cs [hunk 2]
```diff
@@ -333,7 +335,26 @@ internal async Task ProcessMessage(Message receivedMessage, CancellationToken me
                         }
                         else
                         {
-                            transportMessage = JsonSerializer.Deserialize<TransportMessage>(receivedMessage.Body, transportMessageSerializerOptions);
+                            if (nativeHeadersContainNServiceBusMessageIdHeader)
+                            {
+                                // Assume sender is an NServiceBus endpoint when there is the NServiceBus.MessageId header in the native SQS message headers
+                                // HINT: There is no way to know if this is a transport message or not without trying to deserialize it
```
Comment (mauroservienti) [inline src/NServiceBus.Transport.SQS/InputQueuePump.cs:341]:
```suggestion
```

File: src/NServiceBus.Transport.SQS/TransportMessage.cs [hunk 3]
```diff
@@ -40,6 +40,7 @@ public TransportMessage(OutgoingMessage outgoingMessage, DispatchProperties prop
             }
         }
 
+        [System.Text.Json.Serialization.JsonRequired]
```
Comment (mauroservienti) [inline src/NServiceBus.Transport.SQS/TransportMessage.cs:43]: We can revert this

```suggestion
```

**Comment by SzymonPobiega** [comment]
Replaced by https://github.com/Particular/NServiceBus.AmazonSQS/pull/2341

**Pull request closed without merging**